src/bank.rs

```diff
@@ -632,7 +603,57 @@ impl Bank {
                 return Err(BankError::ProgramRuntimeError(instruction_index as u8));
             }
         } else {
-            self.loaded_contract(tx_program_id, tx, instruction_index, program_accounts)?;
+            let mut depth = 0;
+            let mut keys = Vec::new();
+            let mut accounts = Vec::new();
+
+            let mut program_id = tx.program_ids[instruction_index];
+            loop {
+                if NativeProgram::check_id(&program_id) {
+                    // at root of call chain
+                    break;
+                }
+
+                if depth >= 5 {
+                    return Err(BankError::CallChainTooDeep);
+                }
+                depth += 1;
+
+                // TODO fail gracefully
```

Comment:
can this TODO be fixed before this PR merges?

Reply:
Yes, planning on it